components/script_layout_interface/wrapper_traits.rs

```diff
@@ -308,6 +323,10 @@ pub trait ThreadSafeLayoutNode: Clone + Copy + NodeInfo + PartialEq + Sized {
     /// element style is precomputed, not from general layout itself.
     #[inline]
     fn resolved_style(&self) -> Arc<ServoComputedValues> {
+        if self.is_text_node() {
```

Comment:
Can I do this and below in a followup? I have more patches, one of which switches the query stuff to actually using |element|, but I'd rather not add risk to this PR (since it took me a zillion retries to get a successful try run).